ops.h

```diff
@@ -600,79 +619,60 @@ static HWY_INLINE HWY_MAYBE_UNUSED void MulByConstAndAdd(
   MulByConstAndAdd(c, x, out, size, size);
 }
 
-static HWY_NOINLINE void Softmax(float* HWY_RESTRICT x, size_t size,
-                                 size_t mask_pos) {
+static HWY_NOINLINE void Softmax(float* HWY_RESTRICT x, const size_t size,
+                                 const size_t mask_pos) {
   HWY_DASSERT(size != 0);
   HWY_DASSERT(mask_pos <= size);
 
   namespace hn = hwy::HWY_NAMESPACE;
   using D = hn::ScalableTag<float>;
   const D d;
-  const size_t N = hn::Lanes(d);
 
-  // Find max so we can subtract it below. Avoid hn::Foreach because SVE vectors
-  // cannot be lambda-captured.
-  // TODO(janwas): could be replaced with an hn::Accumulate algo.
-  const hn::Vec<D> vmin = hn::Set(d, hwy::LowestValue<float>());
-  hn::Vec<D> vmax = vmin;
-  size_t idx = 0;
-  if (mask_pos >= N) {
-    for (; idx <= mask_pos - N; idx += N) {
-      vmax = hn::Max(vmax, LoadU(d, x + idx));
-    }
-  }
-  vmax = hn::Max(vmax, LoadNOr(vmin, d, x + idx, mask_pos - idx));
-  vmax = hn::MaxOfLanes(d, vmax);  // broadcast
+  const auto vmin = hn::Set(d, hwy::LowestValue<float>());
+  auto vmax = vmin;
+  Foreach(d, x, mask_pos, vmin,
+          [&vmax](const auto d, const auto value)
+              HWY_ATTR { vmax = hn::Max(vmax, value); });
```

Comment:
@pchx noticed we are missing the reduction set here - after Foreach, broadcasting the max to all lanes:
`vmax = hn::MaxOfLanes(d, vmax); // broadcast`

Reply:
Thanks @pchx and @jan-wassenberg.
In my local test I have noticed ReduceMax is missing.
I will fix it.